ci/scan_deps.py

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env python3
+#
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+#
+# Usage: scan_deps.py --osv-lockfile <lockfile> --output <parsed lockfile>
+#
+# This script parses the dependencies provided in lockfile format for
+# osv-scanner so that the common ancestor commits from the mirrored and
+# upstream for each dependency are provided in the lockfile
+
+import argparse
+import json
+import os
+import shutil
+import subprocess
+import sys
+from compatibility_helper import byte_str_decode
+
+SCRIPT_DIR = os.path.dirname(sys.argv[0])
+CHECKOUT_ROOT = os.path.realpath(os.path.join(SCRIPT_DIR, '..'))
+DEP_CLONE_DIR = CHECKOUT_ROOT + '/clone-test'
+DEPS = os.path.join(CHECKOUT_ROOT, 'DEPS')
+UPSTREAM_PREFIX = 'upstream_'
+
+failed_deps = []  # deps which fail to be cloned or git-merge based
+
+
+def parse_deps_file(lockfile, output_file):
+  """
+  Takes input of fully qualified dependencies,
+  for each dep find the common ancestor commit SHA
+  from the upstream and query OSV API using that SHA
+
+  If the commit cannot be found or the dep cannot be
+  compared to an upstream, prints list of those deps
+  """
+  deps_list = []
+  with open(DEPS, 'r') as file:
+    local_scope = {}
+    global_scope = {'Var': lambda x: x}  # dummy lambda
+    # Read the content.
+    deps_content = file.read()
+
+    # Eval the content.
+    exec(deps_content, global_scope, local_scope)
+
+    # Extract the deps and filter.
+    deps_list = local_scope.get('vars')
+
+  with open(lockfile, 'r') as file:
+    data = json.load(file)
+
+  results = data['results']
+
+  if not os.path.exists(DEP_CLONE_DIR):
+    os.mkdir(DEP_CLONE_DIR)  #clone deps with upstream into temporary dir
```

Comment:
nit: Here and throughout the file. Comments should begin with a capital letter and end with a period. There should be a space between the `#` and the first letter of the comment.

Reply:
Addressed this and other comments